### PR TITLE
Update parser for processing links into arrays

### PR DESCRIPTION
**Pull request by blankdots**
### Description

the structure involved creating unnecessary nesting of links into dict with a list of dict of dict

this aims to simplify it by creating a dict with a objects and a list of objects which flattens the structure.

previously the structure was like:
```
"sampleLinks": {
      "sampleLink": [
        {
          "urlLink": {
            "label": "DNA source",
            "url": "http://ccr.coriell.org/Sections/Collections/NHGRI/hapmap.aspx?PgId=266"
          }
        },
        {
          "xrefLink": {
            "db": "DNA source",
            "id": "1289"
          }
        },
        {
          "xrefLink": {
            "db": "DNA 1 source",
            "id": "12890"
          }
        }
      ]
    }
```

now the structure looks like:
```
"sampleLinks": {
      "urlLink": [
        {
          "label": "DNA source",
          "url": "http://ccr.coriell.org/Sections/Collections/NHGRI/hapmap.aspx?PgId=266"
        }
      ],
      "xrefLink": [
        {
          "db": "DNA source",
          "id": "1289"
        },
        {
          "db": "DNA 1 source",
          "id": "12890"
        }
      ]
    }
```

### Type of change

- [x] New feature (non-breaking change which adds functionality)

### Changes Made

1. split parser
2. flatten structure
3. modify tests.
4. `existingStudyType` was remove to reflect better on the UI use of JSON schemas.

### Testing

- [x] Tests do not apply

### Mentions

Mccabe complexity was 20, so I split in order to reduce it

### CODE REVIEW

File: metadata_backend/helpers/parser.py [hunk 1]
```diff
@@ -40,6 +41,63 @@ def __init__(self,
...
+                 'experimentLinks', 'analysisLinks', 'projectLinks',
+                 'policyLinks', 'dacLinks', 'datasetLinks',
+                 'assemblyLinks', 'submissionLinks']
+
+        children = self.dict()
+        for key, value, _ in self.map_content(data.content):
+            key = self._to_camel(key.lower())
+
+            # we flatten the attributes structure into an list of dict
+            if "Attributes" in key and len(value) == 1:
```
Comment (otahontas) [inline metadata_backend/helpers/parser.py:60]: For less erroneus matching, maybe use same strategy here than for links? 
```
atrributes = ['studyAttributes', 'sampleAttributes', 'runAttributes',
                 'experimentAttributes', 'analysisAttributes', 'projectAttributes',
                 'policyAttributes', 'dacAttributes', 'datasetAttributes',
                 'assemblyAttributes', 'submissionAttributes']
if key in attributes:...
```
etc.

File: metadata_backend/helpers/parser.py [hunk 2]
```diff
@@ -59,7 +117,6 @@ def element_decode(self,
           when there are multiple children with same name - then to list.
         - All "accession" keys are converted to "accesionId", key used by
           this program
-
         Corner cases:
```
Comment (otahontas) [inline metadata_backend/helpers/parser.py:120]: For consistency, we should add `links` and `studyType` related corner cases here. Or, if we rather want to have them as inline comments, move these corner cases from docstrings to inline comments. I'm fine with both ways, but not having spesific comments both in dosctring and as inline comments.

File: metadata_backend/helpers/parser.py (hunk 1 above)
Comment (blankdots) [reply metadata_backend/helpers/parser.py]: done.

File: metadata_backend/helpers/parser.py (hunk 2 above)
Comment (blankdots) [reply metadata_backend/helpers/parser.py:123]: done. added corner cases and removed in-line comments

**Pull request merged**